pkg/kubelet/dockertools/manager.go

```diff
@@ -1013,13 +1013,19 @@ func (dm *DockerManager) getSecurityOpt(pod *api.Pod, ctrName string) ([]string,
 		return nil, nil
 	}
 
-	if !strings.HasPrefix(profile, "localhost") {
+	if !strings.HasPrefix(profile, "localhost/") {
 		return nil, fmt.Errorf("unknown seccomp profile option: %s", profile)
 	}
 
-	file, err := ioutil.ReadFile(filepath.Join(dm.seccompProfileRoot, strings.TrimPrefix(profile, "localhost/")))
+	name := strings.TrimPrefix(profile, "localhost/")
+	cleanName := strings.TrimPrefix(path.Clean("/"+name), "/")
+	if name != cleanName {
+		return nil, fmt.Errorf("invalid seccomp profile name: %s", name)
```

Comment:
In fact, the validation logic belongs into api/validation. I have added a commit implementing that. It's purely additional validation code, nothing old changed. I hope that's fine for 1.3. If not, I can split it and leave the upper code as it is, with an addition PR for post-1.3.